Source/BusinessLogic/Logic/PlayedGames/PlayedGameRetriever.cs

```diff
@@ -173,6 +177,21 @@ public List<PlayedGameSearchResult> SearchPlayedGames(PlayedGameFilter playedGam
                     }).ToList()
                 };
 
+            if (playedGameFilter.IncludeCurrentPlayerGamesOnly)
+            {
+	    	if (currentUser == null)
+		{
+			throw new ArgumentNullException(nameof(currentUser),
+                                      "Must have a valid ApplicationUser if attempting to filter played games to only games for the current user.");
+		}
+
+		var currentUserPlayer = _playerRetriever.GetPlayerForCurrentUser(currentUser.Id, currentUser.CurrentGamingGroupId);
+		if (currentUserPlayer != null)
```

Comment:
One last thing I noticed. If there is no Player associated with the current user, then we should probably just return an empty list rather than not filtering at all. In other words, "It_Should_Return_An_Empty_List_If_Results_Are_Filtered_To_The_Current_Users_Player_But_The_Current_User_Has_No_Player"